src/rt/util/utf.d

```diff
@@ -723,8 +723,8 @@ wstring toUTF16(in char[] s)
     wchar[] r;
     size_t slen = s.length;
 
-    r.length = slen;
-    r.length = 0;
+    if (!__ctfe)
+        r.reserve(slen);
```

Comment:
reserve still does a lot if slen is zero. Maybe better to return early for that case. Same below.

Reply:
Hm... reserve probably shouldn't be expensive if request is 0.

Reply:
Probably can just return `null` in that case, or ['\0'].ptr for z versions.

Reply:
> reserve probably shouldn't be expensive if request is 0

reserve returns the allocated block size, so unfortunately, hard to avoid looking up block info.

Reply:
Okay will add a check for slen == 0